src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -130,7 +129,7 @@ public Result CalculateBasedlinedResult(DictionaryMergeBehavior propertyBagMerge
             // Result exists.
             Result result = CurrentResult.Result.DeepClone();
             result.CorrelationGuid = PreviousResult.Result.CorrelationGuid;
-            result.Suppressions = PreviousResult.Result.Suppressions;
+            result.Suppressions = CurrentResult.Result.Suppressions;
```

Comment:
>Suppressions [](start = 55, length = 12)

Just curious why the DeepClone on line 130 doesn't copy over the suppressions.

Reply:
It does, good catch. I removed this line. You only need to be explicit about the things that _don't_ come from `CurrentResult`. And frankly I'm curious why it takes `CorrelationGuid` from the previous result. The spec says that `CorrelationGuid` groups results into "equivalence classes"; that is, two results are "logically identical" if they have the same `CorrelationGuid` and "logically distinct" if they have different `CorrelationGuid`s. So the matcher should never associate two results with different `CorrelationGuids`. I filed #1616, "Result matcher should never associate results with different CorrelationGuids". 

---
In reply to: [311116348](https://github.com/microsoft/sarif-sdk/pull/1615#discussion_r311116348) [](ancestors = 311116348)